hal/src/main/native/athena/FRCDriverStation.cpp

```diff
@@ -230,6 +230,49 @@ double HAL_GetMatchTime(int32_t* status) {
   return matchTime;
 }
 
+int HAL_GetMatchInfo(HAL_MatchInfo* info) {
+  uint16_t gameSpecificMessageSize = 0;
+  int status = FRC_NetworkCommunication_getMatchInfo(
+      nullptr, nullptr, nullptr, nullptr, nullptr, &gameSpecificMessageSize);
+  if (status < 0) {
+    info->eventName = nullptr;
+    info->gameSpecificMessage = nullptr;
+    info->gameSpecificMessageSize = 0;
+    return status;
+  }
+  info->eventName = static_cast<char*>(std::malloc(256));
+  info->gameSpecificMessage =
+      static_cast<uint8_t*>(std::malloc(gameSpecificMessageSize));
```

Comment:
Allocate +1 and nul terminate it here.

Reply:
See response above for why I don't think this is a guaranteed contract.

Reply:
Ok I'll treat it like a string, and turn it into a `char*` to match event name. We can always change it pretty easily later.